pages/content-ui/vite.config.mts

```diff
@@ -1,24 +1,38 @@
 import { resolve } from 'node:path';
-import { makeEntryPointPlugin } from '@extension/hmr';
 import { isDev, withPageConfig } from '@extension/vite-config';
+import { processCssPlugin } from '@extension/hmr';
+import { getEntryPoints } from './script/entryPoints';
 
 const rootDir = resolve(__dirname);
 const srcDir = resolve(rootDir, 'src');
 
+/**
+ * Here is the place to add new entry points, u can straight away add new folders here to include
+ * without creating folders in src dir manually.
+ * @param {string[]} options.include - Folders to include !!MODIFY ME!!
+ * @param {string[]} options.exclude - Folders to exclude !!MODIFY ME!!
+ * @param {boolean} options.createRecursively - Create folders recursively
+ * @returns {Record<string, string>}
+ */
+const entryPoints = getEntryPoints({ 
+  include: ['page1', 'page2'], createRecursively: true 
+});
+
 export default withPageConfig({
   resolve: {
     alias: {
       '@src': srcDir,
     },
   },
-  plugins: [isDev && makeEntryPointPlugin()],
+  plugins: [processCssPlugin()],
   publicDir: resolve(rootDir, 'public'),
   build: {
     lib: {
-      entry: resolve(srcDir, 'index.tsx'),
+      entry: entryPoints,
       name: 'contentUI',
-      formats: ['iife'],
-      fileName: 'index',
+      formats: ['es'],
+      fileName: (format, entryName) =>
+        isDev ? `${entryName}/index_dev.js` : `${entryName}/index.js`
```

Comment:
Why are you need to have 2 different files?

Reply:
filename should be different in dev and production mode, perhaps hashes should be applied in production? that's the convention no?

Reply:
No, i don't think it's really necessary to add prefix
If there's some differences between dev and prod versions, let's add if statement or sth inside code, and you'll have other output file.